alot/crypto.py

```diff
@@ -249,14 +240,34 @@ def _decrypt_verify_session_keys(encrypted, session_keys):
         ctx = gpg.core.Context()
         ctx.set_ctx_flag("override-session-key", key)
         try:
-            (plaintext, _, verify_result) = ctx.decrypt(
-                    encrypted, verify=True)
-        except gpg.errors.GPGMEError as e:
+            return _decryp_verify_with_context(ctx, encrypted)
+        except GPGProblem:
             continue
-        return verify_result.signatures, plaintext
     raise GPGProblem("No valid session key", code=GPGCode.NOT_FOUND)
 
 
+def _decryp_verify_with_context(ctx, encrypted):
+    """Decrypts the given ciphertext string using the gpg context
+    and returns both the signatures (if any) and the plaintext.
+
+    :param gpg.Context ctx: the gpg context
+    :param bytes encrypted: the mail to decrypt
+    :returns: the signatures and decrypted plaintext data
+    :rtype: tuple[list[gpg.resuit.Signature], str]
+    :raises: :class:`~alot.errors.GPGProblem` if the decryption fails
```

Comment:
sphinx `raises` takes the type as an argument:
`:raises alot.errors.GPGProblem: if the decryption fails`